offixsdk/src/main/java/org/aerogear/offixsdk/worker/OfflineMutationsWorker.kt

```diff
@@ -0,0 +1,115 @@
+package org.aerogear.offixsdk.worker
+
+import android.content.Context
+import android.util.Log
+import androidx.work.Worker
+import androidx.work.WorkerParameters
+import com.apollographql.apollo.ApolloCall
+import com.apollographql.apollo.api.Input
+import com.apollographql.apollo.api.Mutation
+import com.apollographql.apollo.api.Operation
+import com.apollographql.apollo.api.Response
+import com.apollographql.apollo.exception.ApolloException
+import org.aerogear.offixsdk.Application
+import org.aerogear.offixsdk.ClientGenerator
+
+
+/**
+ * Workmanager that replicates the mutations to the server that are stored in the database.
+ * Post firing the mutation, it is deleted from the db
+ */
+
+class OfflineMutationsWorker(context: Context, workerParameters: WorkerParameters) : Worker(context, workerParameters) {
+
+
+    private val getMutationDao = Application.database.mutationDao()
+    private val getUrlDao = Application.database.urlDao()
+
+    val TAG = javaClass.simpleName
+
+    override fun doWork(): Result {
+
+        Log.e(TAG, "In Worker")
+
+        //get the list of mutations
+
+        val listOfMutations = getMutationDao.getAllMutations()
+        Log.e(TAG, "Size of list : ${listOfMutations.size}")
+
+        //get the mutation one by one from database
+
+        listOfMutations.forEach { storedmutation ->
+
+            //Get the class name of the mutation to which it has to be mapped
+            val responseClassName = storedmutation.responseClassName
+
+            val classReflected: Class<*> = Class.forName(responseClassName)
+
+            //Get the constructor of the class, and as apollo generated classes have only one constructor, so take the first one.
+            val constructor = classReflected.constructors.first()
+
+            //Get the parameterTypes
+            val parameters = constructor.parameterTypes
+
+            val jsonValues = arrayListOf<Any>()
+
+            //Get the json object i.e the variables map given as input by the user
+            val jsonObj = storedmutation.valuemap
+
+            //Put all the json values into a list
+            val iter = jsonObj.keys()
+            iter.forEach { key ->
+                jsonValues.add(jsonObj.get(key))
+            }
+
+            Log.e("jsonValuesList ", " ${jsonValues.size}")
```

Comment:
Ok.. Let's review every logger and think about levels we use. Error for everything is ok in the beginning :)